src/main/java/seedu/address/model/consultation/Date.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.consultation;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+/**
+ * Represents a Date in the system.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class Date {
+
+    public static final String MESSAGE_CONSTRAINTS = "Dates should be in the format YYYY-MM-DD, "
+            + "and must be a valid date (e.g. no month 13 or day 32).";
```

Comment:
Consider adding an example of a valid input. User may not understand what "YYYY-MM-DD" means. This helps us to avoid ambiguity.